script.js

```diff
@@ -0,0 +1,89 @@
+//create 3 by 3 board
+$(document).ready(function(){
+  // $(function change_cell() {
+      $("td").on("click", function () {
+        var cell_id = $(this).attr('id');
+        console.log(cell_id);
+        console.log($(this));
+        var table = $(this).closest('table');
+        var currentValue = $(this).html();
+        alert(currentValue);
+
+// allow to input X or O
+      var changeValue = prompt("Enter X or O). (Current " + currentValue + ")");
+        $(this).html(changeValue);
+        if (changeValue !== null){
+          $table.find('[id="' + cell_id + '"]').text(changeValue);
+        }
+  });
+});
+
+grid = [
+    [ "" ,"", "" ]
+    [ "", "", "" ]
+    [ "", "", "" ]
+    ];
+
+var row 1 = grid[0]
+var row 2 = grid[1]
+var row 3 = grid[2]
+
+console.log("row1", row1);
+console.log("row2", row2);
+console.log("row3", row3);
+
+var winning_letter = "none";
+
+//compare first row
+if ((row1[0]) == row1[1] && (row1[1] == row1[2]) && (row1[0] == row1[2])) {
+    winning_letter = row1[0];
+}
+
+//compare second row
+if ((row2[0]) == row2[1] && (row2[1] == row2[2]) && (row2[0] == row2[2])){
+    winning_letter = row2[0];
+}
+
+//compare third row
+if ((row1[0]) == row1[1] && (row1[1] == row1[2]) && (row1[0] == row1[2])){
```

Comment:
see any duplication here?  It looks like you are repeatedly checking to see if 3 cells are equal.  If they are, you are returning a value in any of them (represented by the first item).  That sounds like a great time to extract a function.